### DISCUSSION TITLE
"Analyzed full hash of ... / 965 files" while 3 folders contain ~ 100k files

### DISCUSSION THREAD

**Discussion by staaash** [qarmin/czkawka#904] [Q&A]
I've got three folders containing each some 20-30k fotos. Searching for duplicates (with or without cache enabled) is a matter of seconds, during which it says

"Analyzed full hash of ... / 965 files."

It finds no duplicates (which is okay, if true). If I manually "create" a duplicate, it finds it as well.

What irks / confuses me is that it says "965 files" - rather than ~100k files. How can I be sure that all files were checked/compared?

When doing the full hash dupe check on so many files, it took **hours** in the past, not seconds or minutes. Again, with cache disabled in settings, it takes a few seconds more. But time it takes aside, if the three folders contain >100k files, why does it claim to only check <1k of these files?

PS: Corrolary: Is $all_files below all files checked and compared - or only a (random?) subset for which a hash it analysed:

`progress_analyzed_full_hash = Analyzed full hash of {$file_checked}/{$all_files} files`

from https://fossies.org/linux/czkawka/czkawka_gui/i18n/en/czkawka_gui.ftl

btw, fdupes (in the command line) says:

`Progress [ ... / 74918] 25%`

and find no duplicates either. Still, I'm more happy with having the correct count as 74918, rather than 965...

**Comment by STPKITT** [comment]
Hi,
as I understand it according to https://github.com/qarmin/czkawka/blob/master/instructions/Instruction.md which you should read (& which should be linked prominently instead of just under "Contributions" and hidden in the file structure) the full hash scan is skipped when the size of the compared files and the partly hash are already different, making it needless to also calculate the full hash of those files. That way a whole lot of comparison work is spared.
Greets!
